### PR TITLE
add validation and CustomGlobalExceptionHandler

### PR DESCRIPTION
**Pull request by padu12**

### CODE REVIEW

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
```
Comment (satori11111) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:17]: isbn field has rules to check if it is valid, please create your custom annotation and validator or use annotation "@ISBN" that in org.hibernate.validator.constraints.ISBN

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+    @NotNull
     private BigDecimal price;
```
Comment (satori11111) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:20]: price cannot be less than zero, check it

**Review by satori11111 — CHANGES_REQUESTED** [review]
Great, please look at comments and add improvements

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.validator.constraints.ISBN;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
```
Comment (vovasalyha) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:12]: Most likely, you have some restriction on length of title and author in the database, so put according validation for length. Also you can set some meaningful minimum length, but it's for you to decide, maybe there can be some book with one symbol name, but not empty for sure.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**